### PR TITLE
[Keymap] Add paulomp90 lily58 keymap

### PR DESCRIPTION
**Pull request by paulomp90**
## Description

Add my custom lily58 keymap with Portuguese layout as well as, a custom feature in the shift keys, to move 1 word left <> right when tap, and act as the shift when held

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/lily58/keymaps/paulomp90/config.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#pragma once
```
Comment (waffle87) [inline keyboards/lily58/keymaps/paulomp90/config.h:1]: This file is missing a license header.

File: keyboards/lily58/keymaps/paulomp90/keymap.c [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+        RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+        RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+                                    _______, _______, _______, _______, _______,  _______, _______, _______
+    )
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
```
Comment (waffle87) [inline keyboards/lily58/keymaps/paulomp90/keymap.c:131]: The core [tri layer](https://docs.qmk.fm/#/feature_tri_layer) feature should be used here instead.
```suggestion
```

File: keyboards/lily58/keymaps/paulomp90/config.h (hunk 1 above)
Comment (paulomp90) [reply keyboards/lily58/keymaps/paulomp90/config.h:17]: Done

File: keyboards/lily58/keymaps/paulomp90/keymap.c [hunk 3]
```diff
@@ -0,0 +1,237 @@
+#include QMK_KEYBOARD_H
```
[the remark this answers is not in the document]
Comment (paulomp90) [reply keyboards/lily58/keymaps/paulomp90/keymap.c:17]: done

File: keyboards/lily58/keymaps/paulomp90/keymap.c (hunk 2 above)
Comment (paulomp90) [reply keyboards/lily58/keymaps/paulomp90/keymap.c]: Applied, thanks for your suggestion

**Review by waffle87 — APPROVED** [review]

File: keyboards/lily58/keymaps/paulomp90/config.h [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+#define RGBLIGHT_EFFECT_RAINBOW_MOOD
+#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+#define RGBLIGHT_EFFECT_SNAKE
+#define RGBLIGHT_EFFECT_KNIGHT
+#define RGBLIGHT_EFFECT_CHRISTMAS
+#define RGBLIGHT_EFFECT_STATIC_GRADIENT
+#define RGBLIGHT_EFFECT_RGB_TEST
+#define RGBLIGHT_EFFECT_ALTERNATING
+#define RGBLIGHT_EFFECT_TWINKLE
+#define RGBLED_NUM 27
```
Comment (drashna) [inline keyboards/lily58/keymaps/paulomp90/config.h:35]: This is for the r2g board?

Comment (paulomp90) [reply keyboards/lily58/keymaps/paulomp90/config.h:35]: Yes, but I have some doubts about these `rgblight` options.

Are they unnecessary?

Comment (drashna) [reply keyboards/lily58/keymaps/paulomp90/config.h:35]: They shouldn't be necessary.  And by default, the r2g uses RGB matrix (per key) for lighting, not RGB Light (underglow).  

And in general, these settings are wrong for how the board's hardware is configured, anyways.  So you could safely remove the entire block of RGB Light settings.

Comment (paulomp90) [reply keyboards/lily58/keymaps/paulomp90/config.h]: @drashna removed, thank you very much 🙇‍♂️

File: keyboards/lily58/keymaps/paulomp90/config.h [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define MASTER_LEFT
```
Comment (drashna) [inline keyboards/lily58/keymaps/paulomp90/config.h:19]: Master left isn't needed here, either.  The default behavior is master left, and defaults to that if no other handedness defines are present. 

```suggestion
```

Though, personally, I prefer ee hands, if split pin isn't supported, as either side can be used, that way.

**Review by drashna — APPROVED** [review]

File: keyboards/lily58/keymaps/paulomp90/config.h (hunk 5 above)
Comment (paulomp90) [reply keyboards/lily58/keymaps/paulomp90/config.h]: Thanks @drashna

**Pull request merged**